exasol_advanced_analytics_framework/query_handler/context/top_level_query_handler_context.py

```diff
@@ -163,13 +184,20 @@ def _invalidate(self):
         self._valid_object_proxies = set()
         self._owned_object_proxies = set()
         self._is_valid = False
-        child_context_were_not_released = False
+        not_released_child_contexts = []
+        exceptions_from_not_released_child_contexts = []
         for child_query_handler_context in self._child_query_handler_context_list:
             if child_query_handler_context._is_valid:
```

Comment:
Is the logic of "valid" and "not released" a synonym here? If so, you might want to name one of the two.

Reply:
I will create an issue for it, I don't want to pollute this PR with it.

Reply:
https://github.com/exasol/advanced-analytics-framework/issues/72